config/webpack/webpack.common.ts

```diff
@@ -42,12 +46,8 @@ function mapEnvToLogoSuffix(envFile) {
 
 /**
  * Get a production grade config for web or desktop
- * @param {Object} env
- * @param {String} env.envFile path to the env file to be used
- * @param {'web'|'desktop'} env.platform
- * @returns {Configuration}
  */
-const webpackConfig = ({envFile = '.env', platform = 'web'}) => ({
+const getCommonConfig = ({envFile = '.env', platform = 'web'}): Configuration => ({
```

Comment:
I created a return type for config functions, that's not what you meant?

Reply:
What I meant is to define the type for the function e.g.
```js
type GetConfig = (envFile: EnvFile) => Configuration;
```

Then use that type when defining the functions e.g.
```js
const getCommonConfig: GetConfig = ({envFile = '.env', platform = 'web'}) => ({
```

Though I see multiple return options (`Configuration | Configuration[] | Promise<Configuration>;`) so this may need more work than expected.

Reply:
Yes, it's not exactly the same for these different files. Do you think it still makes sense to create one type?

Reply:
I don't think that would add much value in this case. I added review comments to add types to both the input and the output of each function